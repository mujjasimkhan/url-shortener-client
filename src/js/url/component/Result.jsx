import React from "react";
import { Row, Col, Alert } from 'react-bootstrap';

class Result extends React.Component {
    constructor(props) {
        super(props);
    }
    render() {
        const { shortenedUrlResponse } = this.props;
        return (
            <div>
                    <Row>
                        <Col xs lg="1"></Col>
                        <Col xs lg="5">
                            <Row>
                                <Col>Original URL</Col>
                            </Row>
                            <Row>
                                <Col>{shortenedUrlResponse.url}</Col>
                            </Row>
                        </Col>
                        <Col xs lg="6">
                            <Row>
                                <Col>Shortened UR</Col>
                            </Row>
                            <Row>
                                <Col><a href={"https://"+shortenedUrlResponse.url} target="_blank">{shortenedUrlResponse.shortenedUrl}</a></Col>
                            </Row>
                        </Col>
                    </Row>
            </div>)
    }
}
export default Result;