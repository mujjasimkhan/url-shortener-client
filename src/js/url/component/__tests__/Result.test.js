import React from 'react';
import renderer from 'react-test-renderer';
import Result from '../Result.jsx';
const mockShortenedUrlResponse = {
    'id' : 1,
    'url': "Mock Url",
    'shortenedUrl' : 'www.mockurl.com'
}

it('renders Result components', () => {
  const tree = renderer
    .create(<Result shortenedUrlResponse={mockShortenedUrlResponse} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});