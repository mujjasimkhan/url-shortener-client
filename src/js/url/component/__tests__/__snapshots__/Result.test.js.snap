// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Result components 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-lg-1 col"
    />
    <div
      className="col-lg-5 col"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          Original URL
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          Mock Url
        </div>
      </div>
    </div>
    <div
      className="col-lg-6 col"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          Shortened UR
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          <a
            href="https://Mock Url"
            target="_blank"
          >
            www.mockurl.com
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
